src/net/p2p.rs

```diff
@@ -2833,7 +2862,8 @@ impl PeerNetwork {
             reward_cycle_finish
         );
 
-        for reward_cycle in reward_cycle_start..reward_cycle_finish {
+        // go from latest to earliest reward cycle
+        for reward_cycle in (reward_cycle_finish..reward_cycle_start).rev() {
```

Comment:
Ah, but `reward_cycle_finish <= reward_cycle_start`.  The anti-entropy protocol works backwards, from latest reward cycle to earliest reward cycle.

Reply:
if `reward_cycle_finish <= reward_cycle_start`, wouldn't that make `reward_cycle_start` the latest reward cycle?

Reply:
Yes, it would.  Here's an execution trace of some of the debug statements from a unit test:

```
$ grep -E 'Local blocks inventory|over reward cycles' /tmp/test.out
DEBG [1643228629.268898] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 6-5
DEBG [1643228629.273409] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 5 is BlocksInvData { bitlen: 5, block_bitvec: [31], microblocks_bitvec: [30] }
DEBG [1643228630.151754] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 6-5
DEBG [1643228630.155805] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 5 is BlocksInvData { bitlen: 5, block_bitvec: [3], microblocks_bitvec: [2] }
DEBG [1643228631.350438] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 5-4
DEBG [1643228631.353977] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 4 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228632.425082] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 5-4
DEBG [1643228632.428814] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 4 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228633.348980] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 4-3
DEBG [1643228633.352538] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 3 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228634.376591] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 4-3
DEBG [1643228634.380029] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 3 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228635.396177] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 3-2
DEBG [1643228635.399907] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 2 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228636.350704] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 3-2
DEBG [1643228636.354618] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 2 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228637.306919] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 2-1
DEBG [1643228637.310311] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 1 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228638.418983] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 2-1
DEBG [1643228638.423283] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 1 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228639.353688] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 1-0
DEBG [1643228639.357670] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 0 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228640.394823] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 1-0
DEBG [1643228640.399527] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 0 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228641.443450] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 6-5
DEBG [1643228641.452464] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 5 is BlocksInvData { bitlen: 5, block_bitvec: [31], microblocks_bitvec: [30] }
DEBG [1643228642.554711] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 6-5
DEBG [1643228642.560304] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 5 is BlocksInvData { bitlen: 5, block_bitvec: [15], microblocks_bitvec: [14] }
DEBG [1643228643.357928] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 5-4
DEBG [1643228643.363271] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 4 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228644.372366] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 5-4
DEBG [1643228644.376326] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 4 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228645.404950] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 4-3
DEBG [1643228645.408651] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 3 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228646.427323] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 4-3
DEBG [1643228646.432209] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 3 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228647.400470] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 3-2
DEBG [1643228647.405313] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 2 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228648.366515] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 3-2
DEBG [1643228648.371189] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 2 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228649.301985] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 2-1
DEBG [1643228649.306562] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 1 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228650.444922] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 2-1
DEBG [1643228650.449579] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 1 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228651.346471] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 1-0
DEBG [1643228651.352485] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 0 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228652.412420] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 1-0
DEBG [1643228652.417380] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4242)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 0 is BlocksInvData { bitlen: 5, block_bitvec: [0], microblocks_bitvec: [0] }
DEBG [1643228653.391679] [src/net/p2p.rs:2857] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: run protocol for 1 neighbors, over reward cycles 6-5
DEBG [1643228653.400161] [src/net/p2p.rs:2879] [ThreadId(2)] local.80000000://(bind=127.0.0.1:4240)(pub=None): AntiEntropy: Local blocks inventory for reward cycle 5 is BlocksInvData { bitlen: 5, block_bitvec: [31], microblocks_bitvec: [30] }
```

As you can see, the system starts from the highest reward cycle, looks for data to push, and on the next pass, starts from the next-highest reward cycle, and so on and so forth until it wraps around.  The idea is that other nodes are most likely to be missing _recent_ blocks, so the antientropy system should prioritize them first.